### PR TITLE
Migrate Fluentd networkpolices to generator

### PR DESCRIPTION
**Pull request by robinAwallace**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

When adding extra plugins to fluentd user, there may need some extra network polices. 
This PR migrates fluentd network polices to the new network policy generator and adds the extra configuration for extra plugins.

...

- Fixes #

#### Information to reviewers

We did not have deny-all before in fluentd-system. But I added it as I think it should be there. 

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [x] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: helmfile.d/stacks/fluentd.yaml [hunk 1]
```diff
@@ -27,8 +45,6 @@ templates:
     {{- end }}
     name: fluentd-aggregator
     needs:
```
Comment (aarnq) [inline helmfile.d/stacks/fluentd.yaml:47]:
```suggestion
    needs:
      - fluentd-system/networkpolicy
      - fluentd-system/podsecuritypolicy
```

File: helmfile.d/stacks/fluentd.yaml [hunk 2]
```diff
@@ -50,8 +66,6 @@ templates:
     {{- end }}
     name: fluentd-forwarder
     needs:
```
Comment (aarnq) [inline helmfile.d/stacks/fluentd.yaml:68]:
```suggestion
    needs:
      - fluentd-system/networkpolicy
      - fluentd-system/podsecuritypolicy
```

File: helmfile.d/stacks/fluentd.yaml [hunk 3]
```diff
@@ -72,8 +86,6 @@ templates:
     namespace: fluentd
     name: fluentd
     needs:
```
Comment (aarnq) [inline helmfile.d/stacks/fluentd.yaml:88]:
```suggestion
    needs:
      - fluentd-system/networkpolicy
      - fluentd-system/podsecuritypolicy
```

File: helmfile.d/stacks/fluentd.yaml [hunk 4]
```diff
@@ -95,8 +107,5 @@ templates:
     chart: charts/log-manager
     version: 0.1.0
     name: log-manager
```
Comment (aarnq) [inline helmfile.d/stacks/fluentd.yaml:109]:
```suggestion
    name: log-manager
      - fluentd-system/networkpolicy
      - fluentd-system/podsecuritypolicy
```

File: helmfile.d/values/networkpolicies/common/fluentd.yaml.gotmpl [hunk 5]
```diff
@@ -0,0 +1,43 @@
+{{- $netpol := .Values.networkPolicies -}}
+
+policies:
+  fluentd-system:
+    all-deny:
+      podSelectorLabels: {}
+      egress: {}
+      ingress: {}
+    forwarder:
```
Comment (aarnq) [inline helmfile.d/values/networkpolicies/common/fluentd.yaml.gotmpl:9]: Style:
```suggestion
      ingress: {}
      
    forwarder:
```

File: helmfile.d/values/networkpolicies/common/fluentd.yaml.gotmpl [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+          peers:
+            - podSelectorLabels:
+                app.kubernetes.io/instance: fluentd-aggregator
+          ports:
+            - tcp: 24224
+      ingress:
+        - rule: ingress-rule-prometheus
+          ports:
+            - tcp: 24231
+    aggregator:
```
Comment (aarnq) [inline helmfile.d/values/networkpolicies/common/fluentd.yaml.gotmpl:25]: Style:
```suggestion
            - tcp: 24231
            
    aggregator:
```
I'd also prefer if the aggregator policy was above the forwarder.

File: helmfile.d/values/networkpolicies/workload/fluentd.yaml.gotmpl [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      egress:
+        - name: fluentd-extra-plugin
+          peers:
+            {{- range $netpol.fluentd.extraPlugin.ips }}
+            - cidr: {{ . }}
+            {{- end }}
+          ports:
+            {{- range $netpol.fluentd.extraPlugin.ports }}
+            - tcp: {{ . }}
+            {{- end }}
```
Comment (aarnq) [inline helmfile.d/values/networkpolicies/workload/fluentd.yaml.gotmpl:59]: Maybe this should be a conditional rule in the regular one? Also just check what happens when ips and ports are empty so it don't render a allow all rule, I don't remember if we guard against it in the generator :sweat_smile:

**Review by aarnq — COMMENTED** [review]
Nice!

**Comment by robinAwallace** [comment]
> Nice!

Thanks André, I have done changes based on your commens :slightly_smiling_face:

File: config/schemas/config.yaml [hunk 8]
```diff
@@ -5997,6 +5997,19 @@ properties:
             title: Network Policies Fluentd Enabled
             type: boolean
             default: true
+          extraOutput:
+            title: Extra Output/Plugin Rule
+            description: |-
+                Configure extra output/plugin egress rules.
```
Comment (aarnq) [inline config/schemas/config.yaml:6003]:
```suggestion
            title: Extra Output Rule
            description: |-
                Configure extra output egress rules.
```
The output/plugin was depending on change of `extraOutput`. :slightly_smiling_face:

**Review by aarnq — APPROVED** [review]
Just a minor thing else LGTM.

**Review by lunkan93 — APPROVED** [review]

**Pull request merged**